docs(genapi): add a langchain integration and update bolt.diy integration using GenAPIs (#4911)
scaleway/docs-content: pages/generative-apis/reference-content/integrating-generative-apis-with-popular-tools.mdx

    ```Python
    llm = init_chat_model("mistral-small-3.1-24b-instruct-2503", model_provider="openai", base_url="https://api.scaleway.ai/v1")
    ```
4. Use the `llm` object and the `tools` list to generate a response to your query with the following code:
    ```python
    query = "What is 3 * 12?"
    # You can also try the following query:
    # query = "What is 42 + 4?"

    messages = [HumanMessage(query)]  # We initialize the messages list with the user's query.

    ai_msg = llm_with_tools.invoke(messages)  # We generate a response to the query.
    messages.append(ai_msg)  # We append the response to the messages list.

    for tool_call in ai_msg.tool_calls:
        selected_tool = {"add": add, "multiply": multiply}[tool_call["name"].lower()]  # Depending on the tool name, we select the appropriate tool.
        tool_msg = selected_tool.invoke(tool_call)  # We invoke the selected tool with the tool call.
        messages.append(tool_msg)  # We append the tool's response to the messages list.

    print(llm_with_tools.invoke(messages).content)  # We print the content of the final response.
    ```
5. Run `tools.py`:

>     ```bash
>     python tools.py
>     ```bash
>     The result of 3 * 12 is 36.
>     ```

Reviewer: ```suggestion
    ```bash
    python tools.py
    ```bash
    The result of 3 * 12 is 36.
    ```
```
there's ``` 2 times, this is impacting the formatting for the rest of the page

Author: Does it look good now? I updated it.

LangChain is a popular library for building AI applications. Scaleway's Generative APIs support LangChain for both inference and embeddings.

### Python

#### Function calling

> 1. Install LangChain and his dependencies.
>     ```bash
>     pip install 'langchain>=0.3.24'
>     pip install 'langchain-core>=0.3.55'
>     pip install 'langchain-openai>=0.3.14'
>     pip install 'langchain-text-splitters>=0.3.8'
>     ```
> 2. Create a file `tools.py` and add the following code to it to import and create the tools examples:

Reviewer 2: ```suggestion
1. Run the following commands to install LangChain and its dependencies:
    ```bash
    pip install 'langchain>=0.3.24'
    pip install 'langchain-core>=0.3.55'
    pip install 'langchain-openai>=0.3.14'
    pip install 'langchain-text-splitters>=0.3.8'
    ```
2. Create a file named `tools.py` and paste the code below into it to import and create the tools examples:
```

  ```Python
  from langchain_core.messages import HumanMessage
  from langchain.chat_models import init_chat_model
  from langchain_core.tools import tool

  @tool
  def add(a: int, b: int) -> int:
      """Adds a and b."""
      return a + b

  @tool
  def multiply(a: int, b: int) -> int:
      """Multiplies a and b."""
      return a * b

  tools = [add, multiply]
  ```

> 3. You can configure the `init_chat_model` function to use Scaleway's Generative APIs.

Reviewer 2: ```suggestion
3. Configure the `init_chat_model` function to use Scaleway's Generative APIs.
```

  ```Python
  llm = init_chat_model("mistral-small-3.1-24b-instruct-2503", model_provider="openai", base_url="https://api.scaleway.ai/v1")
  ```

> 4. Everything is now set up, you can use the `llm` object and the `tools` list to generate a response to a query with the following code:

Reviewer 2: ```suggestion
4. Use the `llm` object and the `tools` list to generate a response to your query with the following code:
```

  ```python
  query = "What is 3 * 12?"
  # You can also try the following query:
  # query = "What is 42 + 4?"

  messages = [HumanMessage(query)]  # We initialize the messages list with the user's query.

  ai_msg = llm_with_tools.invoke(messages)  # We generate a response to the query.
  messages.append(ai_msg)  # We append the response to the messages list.

  for tool_call in ai_msg.tool_calls:
      selected_tool = {"add": add, "multiply": multiply}[tool_call["name"].lower()]  # Depending on the tool name, we select the appropriate tool.
      tool_msg = selected_tool.invoke(tool_call)  # We invoke the selected tool with the tool call.
      messages.append(tool_msg)  # We append the tool's response to the messages list.

  print(llm_with_tools.invoke(messages).content)  # We print the content of the final response.
  ```

> 5. Finally, run the `tools.py`:
>   ```bash
>   python tools.py
>   ```
>   The response should display the result of the calculation:

Reviewer 2: ```suggestion
5. Run `tools.py`:
  ```bash
  python tools.py
  ```
  The result of the calculation should display in the output:
```

5. Click **Local Providers** to add a new external provider configuration.
6. Toggle the switch next to **OpenAILike** to enable it. Then, enter the Scaleway API endpoint: `https://api.scaleway.ai/v1` as the base URL.
7. In Bolt's main menu, select `OpenAILike` and input your **Scaleway Secret Key** as the `OpenAILike API Key`.
8. Select one of the supported models from Scaleway Generative APIs. For best results with Bolt.diy, which requires a significant amount of output tokens (8000 by default), start with the `gemma-3-27b-it` model.
9. Enter your prompt in the Bolt.diy interface to see your application being generated.

<Message type="important">

>     Only models that have a maximum output token of at least 8000 tokens are supported. You can found the list of Generative APIs models [here](https://www.scaleway.com/en/docs/generative-apis/reference-content/supported-models/#chat-models).

Reviewer 2: ```suggestion
    Only models that have a maximum output token of at least 8000 tokens are supported. Refer to the [list of Generative APIs models](/generative-apis/reference-content/supported-models/#chat-models) for more information.
```
